github_connector/models/github_repository_branch.py

```diff
@@ -275,6 +293,66 @@ def _analyze_code(self):
                         branch.name,
                         e,
                     )
+        # github_analysis_rule
+        for branch in self:
+            rule_ids = self.env["github.analysis.rule"].search(
+                [
+                    "|",
+                    (
+                        "id",
+                        "in",
+                        branch.repository_id.organization_id.analysis_rule_ids.mapped(
+                            "analysis_rule_id"
+                        ).ids,
+                    ),
+                    (
+                        "id",
+                        "in",
+                        branch.repository_id.analysis_rule_ids.mapped(
+                            "analysis_rule_id"
+                        ).ids,
+                    ),
+                ]
+            )
+            if rule_ids:
+                for rule_id in rule_ids:
+                    # get + analyse
+                    res = {
+                        "paths": [],
+                        "code": 0,
+                        "documentation": 0,
+                        "empty": 0,
+                        "string": 0,
+                    }
+                    for match in rule_id._get_matches(branch.local_path):
+                        # _logger.info("analyse file %s "% match)
+                        res_file = rule_id._analysis_file(
+                            branch.local_path + "/" + match
+                        )
+                        res["paths"].append(res_file["path"])
+                        # define values
+                        for key in ["code", "documentation", "empty", "string"]:
```

Comment:
Fixed length + Immutable + Less memory usage
```suggestion
                        for key in ("code", "documentation", "empty", "string"):
```